File: python/plants/mlecz.py
from plants.roslina import Roslina


class Mlecz(Roslina):

    def __str__(self):
        return self._species + "[" + str(self._x) + "][" + str(self._y) + "]"

    def __init__(self, x, y, world, sila=0):
        Roslina.__init__(self, "mlecz", sila, 0, x, y, world)

    def akcja(self):
        for x in range(3):
            Roslina.akcja(self)
    def rysowanie(self):
        return "M"

File: java/src/Rosliny/Mlecz.java
package Rosliny;
import Swiat.*;

public class Mlecz extends Roslina{
    public Mlecz(int x, int y, Swiat swiat){
        super(Gatunek.MLECZ, 0, 0, x, y, swiat);
    }
    public Mlecz(int x, int y, int sila, Swiat swiat){
        super(Gatunek.MLECZ, sila, 0, x, y, swiat);
    }

    @Override
    public Polozenie akcja(){
        int czyRozmnozy;
        for(int i = 0; i < 3 ; i++)
        {
            czyRozmnozy = this.swiat.getLosowaLiczba(0, this.szansaNaRozmnozenie);
            if (czyRozmnozy == 0)
                this.swiat.dodajOrganizm(this.gatunek, this.polozenie.x, this.polozenie.y);
        }
        return this.polozenie;
    }

    @Override
    public String rysowanie(){return "M";}
}
